scripts/lib/CIME/utils.py

```diff
@@ -777,6 +777,70 @@ def compute_total_time(job_cost_map, proc_pool):
 
     return current_time
 
+def format_time(time_format, input_format, input_time):
+    """
```

Comment:
Would you be willing to add some doctest type tests here to show usage with various inputs?  I think that would be a great addition here.   Thanks